internal/util/jvm.go

```diff
@@ -47,5 +54,118 @@ func DetectJVMName(executor effect.Executor) (string, error) {
 			break
 		}
 	}
+
+	if err := scanner.Err(); err != nil {
+		return "", fmt.Errorf("unable to read cache ratio\n%w", err)
+	}
+
 	return "", nil
 }
+
+type SharedClassCache struct {
+	Name string
+	Path string
+
+	Executor effect.Executor
+	Logger   bard.Logger
+}
+
+type SharedClassCacheOptions struct {
+	Enabled       bool
+	SizeMB        int
+	NumIterations int
+	Trim          bool
+}
+
+func (scc SharedClassCache) GetFillRatio() (float64, error) {
+	sccOption := fmt.Sprintf("-Xshareclasses:name=%s,cacheDir=%s,printTopLayerStats", scc.Name, scc.Path)
+	args := []string{sccOption, "-version"}
+	buf := &bytes.Buffer{}
+	stderrBuf := &bytes.Buffer{}
+	err := scc.Executor.Execute(effect.Execution{
+		Command: "java",
+		Args:    args,
+		Stdout:  buf,
+		Stderr:  stderrBuf,
+	})
+	var exitError *exec.ExitError
+	if errors.As(err, &exitError) && exitError.ExitCode() != 1 {
+		return 0.0, fmt.Errorf("unable to get cache stats\n%w\n%s", err, stderrBuf.String())
+	}
+
+	r, err := regexp.Compile(`Cache is (\d+)% full`)
+	if err != nil {
+		return 0.0, err
+	}
```

Comment:
```suggestion
	r := regexp.MustCompile(`Cache is (\d+)% full`)
```

You can probably use MustCompile here. The regular expression is a static string, so if it works in testing it should never panic. 

https://pkg.go.dev/regexp#MustCompile